ssl/quic/quic_channel.c

```diff
@@ -113,6 +114,88 @@ static int gen_rand_conn_id(OSSL_LIB_CTX *libctx, size_t len, QUIC_CONN_ID *cid)
     return 1;
 }
 
+static unsigned long chan_reset_token_hash(
+        const QUIC_STATELESS_RESET_TOKEN *tok)
+{
+    unsigned long h;
+
+    assert(sizeof(h) <= sizeof(tok->token));
+    memcpy(&h, tok->token, sizeof(h));
+    return h;
+}
+
+static int chan_reset_token_cmp(const QUIC_STATELESS_RESET_TOKEN *t1,
+                                const QUIC_STATELESS_RESET_TOKEN *t2)
+{
+    return CRYPTO_memcmp(t1->token, t2->token, sizeof(t1->token));
+}
+
+static int reset_token_obfuscate(QUIC_STATELESS_RESET_TOKEN *out,
+                                 const unsigned char *in, size_t len)
+{
+    /*
+     * TODO(QUIC): update this to AES encrypt the token in ECB mode with a
+     * random (per channel) key.
+     */
+    memcpy(out, in, sizeof(*out));
+    return 1;
+}
+
+/*
+ * Remove the old token from the table of legal tokens and insert the new
+ * one in it's place.
+ */
+static int chan_update_reset_token(QUIC_CHANNEL *ch,
+                                   const unsigned char *new, size_t new_len)
+{
+    QUIC_STATELESS_RESET_TOKEN *r;
+
+    /*
+     * Remove any existing reset token, if this fails it means there wasn't
+     * one and this can safely be ignored.
+     */
+    (void)lh_QUIC_STATELESS_RESET_TOKEN_delete(ch->stateless_reset_tokens,
```

Comment:
I suppose cur_remote_reset_token will not be initialised when this is called the first time (from TPARAM processing). Since there will be nothing in the LHASH anyway I guess this in practice will not be an issue...

Reply:
That's the plan.  That alternative is an in use flag as part of the reset token.